### PR TITLE
Redo pr 17 and after

### PR DESCRIPTION
**Pull request by lukaso**
Show event logs in UI

### CODE REVIEW

File: console/src/routes/advanced/+page.svelte [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+		<HopsNav />
+
+		<!--Page title-->
+		<div class="md:pl-20 md:pr-20 mx-8 md:mx-auto mt-12 pb-4">
+			<h1 class="text-left text-5xl font-normal mb-2 dark:text-white">Advanced</h1>
+			<h2 class="text-2xl mb-4 text-grey dark:text-white dark:text-opacity-50">
+				Showing all events since {ago}
+			</h2>
+
+			<!--Tabs-->
```
Comment (lukaso) [inline console/src/routes/advanced/+page.svelte:63]: Reason this is commented out is that the next story is to add this functionality in. Makes it (arguably) easier than stripping it out and then finding it again.

**Comment by lukaso** [comment]
[[MVF:Medium] Hops users can see log of events](https://trello.com/c/SyP9xTso/1021-mvfmedium-hops-users-can-see-log-of-events)

File: nats/msgschema.go [hunk 2]
```diff
@@ -153,6 +155,17 @@ func NewResultMsg(startedAt time.Time, result interface{}, err error) ResultMsg
 	return resultMsg
 }
 
+func EventFilter(accountId string) string {
+	tokens := []string{
+		accountId,
+		"notify",
```
Comment (manterfield) [inline nats/msgschema.go:161]: Use the ChannelNotify const

File: nats/msgschema.go [hunk 3]
```diff
@@ -153,6 +155,17 @@ func NewResultMsg(startedAt time.Time, result interface{}, err error) ResultMsg
...
+		"*",
+		"event",
```
Comment (manterfield) [inline nats/msgschema.go:163]: Create new const at top of file (similar to `HopsMessageId`, e.g. `SourceMessageId`)

File: nats/client.go [hunk 4]
```diff
@@ -214,6 +231,69 @@ func (c *Client) FetchMessageBundle(ctx context.Context, newMsg *MsgMeta) (Messa
...
+		FilterSubjects: []string{EventFilter(c.accountId)},
+		DeliverPolicy:  jetstream.DeliverByStartTimePolicy,
+		OptStartTime:   &start,
+	}
+	cons, err := c.JetStream.OrderedConsumer(ctx, c.accountId, consumerConf)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to create ordered consumer: %w", err)
+	}
+
+	msgs, _ := cons.FetchNoWait(1000000)
```
Comment (manterfield) [inline nats/client.go:252]: Change as discussed

**Review by manterfield — CHANGES_REQUESTED** [review]

File: internal/httpserver/eventrouter.go [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+		c.logger.Error().Err(err).Msg("Error reading event history")
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(eventLog)
+}
+
+func eventLogFromMsgMetas(msgs []*nats.MsgMeta, start time.Time) (*EventLog, error) {
```
Comment (lukaso) [inline internal/httpserver/eventrouter.go:76]:
```suggestion
// eventLogFromMsgMetas converst event messasges to the /events api, returning only the last 100 
func eventLogFromMsgMetas(msgs []*nats.MsgMeta, start time.Time) (*EventLog, error) {
```

**Pull request closed without merging**